test/installers/service_installer.go

```diff
@@ -0,0 +1,287 @@
+		return waitForAllCrsRemoval(si, namespace, waitForAllCrsRemovalTimeout)
+	}); !ok {
+		success = false
+	}
+	if ok := executeOnNamespacedServices(namespace, func(si NamespacedServiceInstaller) bool {
+		return waitForAllCrsRemoval(si, namespace, waitForAllCrsRemovalTimeout)
+	}); !ok {
+		success = false
+	}
+
+	// Delete namespaced services
+	if ok := executeOnNamespacedServices(namespace, func(si NamespacedServiceInstaller) bool {
+		if err := si.uninstallFromNamespace(namespace); err != nil {
+			framework.GetLogger(namespace).Error(err, "Error uninstalling service from namespace")
+			return false
+		}
+		return true
+	}); !ok {
+		success = false
+	}
+
+	return success
+}
+
+// UninstallServicesFromCluster uninstalls the cluster wide service from whole cluster. Returns false in case of any error while uninstalling.
+func UninstallServicesFromCluster() (success bool) {
+	return executeOnClusterWideServices(func(si ClusterWideServiceInstaller) bool {
+		if err := si.uninstallFromCluster(); err != nil {
+			framework.GetMainLogger().Error(err, "Error uninstalling service from cluster")
+			return false
+		}
+		return true
+	})
+}
+
+// YAML cluster wide service specification
+
+// Make sure that YamlClusterWideServiceInstaller can be typed to ClusterWideServiceInstaller
+var _ ClusterWideServiceInstaller = &YamlClusterWideServiceInstaller{}
+
+// YamlClusterWideServiceInstaller installs service using YAML files applied to specific namespace
+type YamlClusterWideServiceInstaller struct {
+	// Install service for all namespaces
+	installClusterYaml func() error
+	// Namespace used for cluster wide service installation.
+	installationNamespace string
+	// Wait until the service is up and running
+	waitForClusterYamlServiceRunning func() error
+	// Return all CRs of this service which exists in this namespace
+	getAllClusterYamlCrsInNamespace func(string) ([]meta.ResourceObject, error)
+	// Uninstall service from whole cluster
+	uninstallClusterYaml func() error
+}
+
+// Install the cluster wide service using YAML files into cloud
+func (installer *YamlClusterWideServiceInstaller) Install(namespace string) error {
+	// Store cluster wide service installer to use for uninstalling purposes
+	if _, loaded := installedClusterWideServices.LoadOrStore(installer, "true"); loaded {
+		// Should be running already, wait until it is up
+		return installer.waitForClusterYamlServiceRunning()
+	}
+
+	monitorNamespace(installer.installationNamespace)
+
+	if err := installer.installClusterYaml(); err != nil {
+		return err
+	}
+
+	return installer.waitForClusterYamlServiceRunning()
+}
+
+func (installer *YamlClusterWideServiceInstaller) getAllCrsInNamespace(namespace string) ([]meta.ResourceObject, error) {
+	return installer.getAllClusterYamlCrsInNamespace(namespace)
+}
+
+func (installer *YamlClusterWideServiceInstaller) uninstallFromCluster() error {
+	stopNamespaceMonitoring(installer.installationNamespace)
```

Comment:
we don't remove here from `installedClusterWideServices` ?

Reply:
Yes we do. That is what I forgot to fix, thanks for reminding :)

Reply:
Removing is done as part of public uninstalling functions.